From a77760833c998e1549e1dde740699b2cb0c55bc5 Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Fri, 4 May 2018 15:20:17 +0800
Subject: [PATCH 1883/2313] media: intel-ipu4: imx319: Rearrange test pattern
 value order

Message for Open Source:
Rearrange test pattern order to make the value aligned with other
sensor drivers.

Message for Internal:
[Issue/Feature] The test pattern value definition in Android
camera HAL api does not match with the value definition in sensor.
[Root Cause/Changes] Rearrange test pattern order to make the value
consistent among sensor drivers and match with the definition in
Android camera HAL.

Change-Id: Ie265588f92b489e281ee15f0c9e2cb2ba5c2ee01
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/i2c/imx319.c | 18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/imx319.c b/drivers/media/i2c/imx319.c
index 4ccd4a3..9f57c4f 100644
--- a/drivers/media/i2c/imx319.c
+++ b/drivers/media/i2c/imx319.c
@@ -50,6 +50,11 @@
 
 /* Test Pattern Control */
 #define IMX319_REG_TEST_PATTERN		0x0600
+#define IMX319_TEST_PATTERN_DISABLED		0
+#define IMX319_TEST_PATTERN_SOLID_COLOR		1
+#define IMX319_TEST_PATTERN_COLOR_BARS		2
+#define IMX319_TEST_PATTERN_GRAY_COLOR_BARS	3
+#define IMX319_TEST_PATTERN_PN9			4
 
 /* Flip Control */
 #define IMX319_REG_ORIENTATION		0x0101
@@ -1591,12 +1596,20 @@ static const struct imx319_reg mode_1280x720_regs[] = {
 
 static const char * const imx319_test_pattern_menu[] = {
 	"Disabled",
-	"Solid color",
 	"100% color bars",
+	"Solid color",
 	"Fade to gray color bars",
 	"PN9"
 };
 
+static const int imx319_test_pattern_val[] = {
+	IMX319_TEST_PATTERN_DISABLED,
+	IMX319_TEST_PATTERN_COLOR_BARS,
+	IMX319_TEST_PATTERN_SOLID_COLOR,
+	IMX319_TEST_PATTERN_GRAY_COLOR_BARS,
+	IMX319_TEST_PATTERN_PN9,
+};
+
 /* Configurations for supported link frequencies */
 /* Menu items for LINK_FREQ V4L2 control */
 static const s64 link_freq_menu_items[] = {
@@ -1869,7 +1882,8 @@ static int imx319_update_digital_gain(struct imx319 *imx319, u32 d_gain)
 static int imx319_enable_test_pattern(struct imx319 *imx319, u32 pattern)
 {
 	return imx319_write_reg(imx319, IMX319_REG_TEST_PATTERN,
-				 IMX319_REG_VALUE_16BIT, pattern);
+				IMX319_REG_VALUE_16BIT,
+				imx319_test_pattern_val[pattern]);
 }
 
 static int imx319_set_ctrl(struct v4l2_ctrl *ctrl)
-- 
2.7.4

